fake_client.go

```diff
@@ -704,6 +714,53 @@ func (c *FakeClient) FindKubernetesCluster(search string) (*KubernetesCluster, e
 	return nil, ZeroMatchesError.wrap(err)
 }
 
+// ListKubernetesClusterInstances implemented in a fake way for automated tests
+func (c *FakeClient) ListKubernetesClusterInstances(id string) ([]Instance, error) {
+	for _, cluster := range c.Clusters {
+		if cluster.ID == id {
+			instaces := make([]Instance, len(cluster.Instances))
+			for idx, instance := range cluster.Instances {
+				ins := Instance{
+					ID:              instance.ID,
+					Hostname:        instance.Hostname,
+					Size:            instance.Size,
+					PublicIP:        instance.PublicIP,
+					Status:          instance.Status,
+					InitialUser:     instance.InitialUser,
+					InitialPassword: instance.InitialPassword,
+					FirewallID:      instance.FirewallID,
+					CPUCores:        instance.CPUCores,
+					RAMMegabytes:    instance.RAMMegabytes,
+					DiskGigabytes:   instance.DiskGigabytes,
+					Region:          instance.Region,
+				}
+				instaces[idx] = ins
+			}
+			return instaces, nil
+		}
+	}
+
+	err := fmt.Errorf("unable to find %s, zero matches", id)
+	return nil, DatabaseKubernetesClusterNotFoundError.wrap(err)
+}
+
+// FindKubernetesClusterInstance implemented in a fake way for automated tests
+func (c *FakeClient) FindKubernetesClusterInstance(clusterID, instanceID string) (*Instance, error) {
+	instances, err := c.ListKubernetesClusterInstances(clusterID)
+	if err != nil {
+		return nil, decodeError(err)
+	}
+
+	for _, instance := range instances {
+		if instance.ID == instanceID {
+			return &instance, nil
+		}
+	}
+
+	err = fmt.Errorf("unable to find %s, zero matches", instanceID)
+	return nil, ZeroMatchesError.wrap(err)
```

Comment:
Same thing here. Can we check for MultipleMatchesError here as well?

Reply:
Actually, since this is checking for ID, we won't need it here but still need it in `FindLoadBalancer` because it checks either by ID or by name.